packages/frontend/src/lib/notification/ContainerConnectionStatusInfo.spec.ts

```diff
@@ -0,0 +1,258 @@
+
+  const banner = screen.getByLabelText('Container connection info banner');
+  expect(banner).toBeInTheDocument();
+  const titleDiv = screen.getByLabelText('title');
+  expect(titleDiv).toBeInTheDocument();
+  expect(titleDiv.textContent).equals('No Podman machine is running');
+  const descriptionDiv = screen.getByLabelText('description');
+  expect(descriptionDiv).toBeInTheDocument();
+  expect(descriptionDiv.textContent).equals('Please start a Podman Machine before proceeding further.');
+
+  const btnStart = screen.getByRole('button', { name: 'Start now' });
+  expect(btnStart).toBeInTheDocument();
+
+  await userEvent.click(btnStart);
+
+  expect(navigateMock).toBeCalled();
+});
+
+test('should show no running machine banner if there is no running machine and no action if canRedirect is disabled', async () => {
+  const noMachineInfo: ContainerConnectionInfo = {
+    status: 'no-machine',
+    canRedirect: false,
+  };
+
+  render(ContainerConnectionStatusInfo, { connectionInfo: noMachineInfo });
+
+  const banner = screen.getByLabelText('Container connection info banner');
+  expect(banner).toBeInTheDocument();
+  const titleDiv = screen.getByLabelText('title');
+  expect(titleDiv).toBeInTheDocument();
+  expect(titleDiv.textContent).equals('No Podman machine is running');
+  const descriptionDiv = screen.getByLabelText('description');
+  expect(descriptionDiv).toBeInTheDocument();
+  expect(descriptionDiv.textContent).equals('Please start a Podman Machine before proceeding further.');
+
+  const btnStart = screen.queryByRole('button', { name: 'Start now' });
+  expect(btnStart).not.toBeInTheDocument();
+});
+
+test('should show lowResourcesMachine banner if the running machine has not enough resources and both canEdit and canRedirect are true', async () => {
+  const navigateMock = vi.spyOn(studioClient, 'navigateToEditConnectionProvider');
+  const connectionInfo: ContainerConnectionInfo = {
+    name: 'Podman Machine',
+    canEdit: true,
+    canRedirect: true,
+    cpus: 4,
+    cpusExpected: 10,
+    memoryExpected: 10,
+    memoryIdle: 5,
+    status: 'low-resources',
+  };
+
+  render(ContainerConnectionStatusInfo, { connectionInfo });
+
+  const banner = screen.getByLabelText('Container connection info banner');
+  expect(banner).toBeInTheDocument();
+  const titleDiv = screen.getByLabelText('title');
+  expect(titleDiv).toBeInTheDocument();
+  expect(titleDiv.textContent).equals('Upgrade your Podman machine for best AI performance');
+  const descriptionDiv = screen.getByLabelText('description');
+  expect(descriptionDiv).toBeInTheDocument();
+  expect(descriptionDiv.textContent).equals(
+    `Your Podman Machine has ${connectionInfo.cpus} vCPUs and ${filesize(connectionInfo.memoryIdle, { base: 2 })} of memory available. We recommend upgrading your Podman machine with at least ${connectionInfo.cpusExpected} vCPUs and ${filesize(connectionInfo.memoryExpected, { base: 2 })} of memory for better AI performance.`,
```

Comment:
```suggestion
    `Your Podman Machine has ${connectionInfo.cpus} vCPUs and ${filesize(connectionInfo.memoryIdle, { base: 2 })} of memory available. We recommend upgrading your Podman Machine to at least ${connectionInfo.cpusExpected} vCPUs and ${filesize(connectionInfo.memoryExpected, { base: 2 })} of memory for better AI performance.`,
```

Should we also add link to the settings menu?  I guess that most users don't know how to upgrade the resources.

Reply:
There is already the button on the right (when we can show it), isn't that enough?

The different scenario are:
1. there is no podman machine running -> there is a `Start now` button that redirects the user to the resources page where they can see all podman machines available and start/create one
2. there is a podman machine but it does not have the resources we recommend ->
  2.1 if on mac -> we show the 'Update now' button that redirects the user to the page to edit the podman machine resources inside podman desktop
  2.2 if on windows -> there is no button as WSL uses the resources from the host and there is no reason to update the podman machine

Reply:
Edit done 👍 I also replaced `Upgrade` with `Update` as Jeff asked for the change

Reply:
I totally missed the button for some reason.  Looks great, thank you!